pkg/frontend/querymiddleware/results_cache.go

```diff
@@ -639,3 +649,16 @@ func cacheHashKey(key string) string {
 	// Hex because memcache keys must be non-whitespace non-control ASCII
 	return hex.EncodeToString(hasher.Sum(nil))
 }
+
+// extentSamplesInSubrange counts the appropriate number of samples for the subrange within the extent.
+// subrangeStart and subrangeEnd must be within the extent.
+func extentSamplesInSubrange(extent Extent, subrangeStart int64, subrangeEnd int64) uint64 {
+	extentTimeRange := extent.End - extent.Start
+	subrange := subrangeEnd - subrangeStart
+	if extentTimeRange > 0 {
```

Comment:
Shouldn't happen, but maybe we should also add a protection on subrange too?

```suggestion
	if extentTimeRange > 0 && subrange > 0 {
```

Reply:
I'll answer in the thread below, to keep conversation about edge cases in one place.